homeassistant/components/frontend/__init__.py

```diff
@@ -281,22 +291,51 @@ def async_setup(hass, config):
 
     repo_path = conf.get(CONF_FRONTEND_REPO)
     is_dev = repo_path is not None
+    hass.data[DATA_JS_VERSION] = js_version = conf.get(CONF_JS_VERSION)
 
     if is_dev:
         hass.http.register_static_path(
             "/home-assistant-polymer", repo_path, False)
         hass.http.register_static_path(
             "/static/translations",
-            os.path.join(repo_path, "build/translations"), False)
-        sw_path = os.path.join(repo_path, "build/service_worker.js")
+            os.path.join(repo_path, "build-translations"), False)
+        sw_path_es5 = os.path.join(repo_path, "build/service_worker.js")
+        sw_path_es6 = os.path.join(repo_path, "build-es6/service_worker.js")
         static_path = os.path.join(repo_path, 'hass_frontend')
     else:
         import hass_frontend
-        frontend_path = hass_frontend.where()
-        sw_path = os.path.join(frontend_path, "service_worker.js")
-        static_path = frontend_path
+        import hass_frontend_es6
+        sw_path_es5 = os.path.join(hass_frontend.where(), "service_worker.js")
+        sw_path_es6 = os.path.join(hass_frontend_es6.where(),
+                                   "service_worker.js")
+        # /static points to es5 dir. However all files that differ between the
+        # dirs are registered separately.
+        static_path = hass_frontend.where()
```

Comment:
we cache `static_path` but still call `where()` in next lines.

Would it be an idea to base this off a dictionary that will loop through es5/es6 variants.

```python
paths = {
  '/static/frontend-{}.html': 'frontend.html',
}

for url_path, file_path in paths.items():
    …
```

Reply:
Done